src/Avalonia.Base/Styling/StyleBase.cs

```diff
@@ -11,7 +11,7 @@ namespace Avalonia.Styling
     /// <summary>
     /// Base class for <see cref="Style"/> and <see cref="ControlTheme"/>.
     /// </summary>
-    public abstract class StyleBase : AvaloniaObject, IStyle, IResourceProvider
+    public abstract class StyleBase : AvaloniaObject, IStyle, IResourceProvider, IAddChild, IAddChild<IStyle>, IAddChild<SetterBase>
```

Comment:
Have you checked if IAddChild<T> is sufficient?

Reply:
Just checked and its actually enough to fix IDE errors appearing, removed the generic ones

https://github.com/AvaloniaUI/Avalonia/pull/18800/commits/393000cf92e16696c7d0a2688a52ec654cd62c6e

Reply:
The generic ones did not help, only adding the IAddChild with object works in IDE.